lib/bolt/puppetdb/client.rb

```diff
@@ -68,6 +85,23 @@ def http_client
         @http
       end
 
+      def reject_url
+        @bad_urls << @current_url if @current_url
+        @current_url = nil
+      end
+
+      def uri
+        @current_url ||= (Set.new(@config.server_urls) - @bad_urls).first
```

Comment:
Sets aren't guaranteed to preserve insertion order. But you can also just subtract arrays, so `@config.server_urls - @bad_urls.to_a` would work.

Reply:
In practice, it probably does preserve insertion order because it's backed by a Hash under the hood, which preserves insertion order. But the specified behavior doesn't make that guarantee, so I would prefer not to rely on it (implementations can change). Order is important, because generally `server_urls` represents primary and secondary.